src/Microsoft.AspNetCore.SignalR.Core/HubEndPoint.cs

```diff
@@ -78,85 +61,38 @@ public class HubEndPoint<THub> : IInvocationBinder where THub : Hub
             {
-                _logger.ErrorProcessingRequest(ex);
+                Log.ErrorProcessingRequest(_logger, ex);
                 await HubOnDisconnectedAsync(connection, ex);
                 throw;
             }
 
             await HubOnDisconnectedAsync(connection, null);
         }
 
-        private async Task HubOnConnectedAsync(HubConnectionContext connection)
-        {
-            try
-            {
-                using (var scope = _serviceScopeFactory.CreateScope())
-                {
-                    var hubActivator = scope.ServiceProvider.GetRequiredService<IHubActivator<THub>>();
-                    var hub = hubActivator.Create();
-                    try
-                    {
-                        InitializeHub(hub, connection);
-                        await hub.OnConnectedAsync();
-                    }
-                    finally
-                    {
-                        hubActivator.Release(hub);
-                    }
-                }
-            }
-            catch (Exception ex)
-            {
-                _logger.ErrorInvokingHubMethod("OnConnectedAsync", ex);
-                throw;
-            }
-        }
-
         private async Task HubOnDisconnectedAsync(HubConnectionContext connection, Exception exception)
         {
+            // We wait on abort to complete, this is so that we can guarantee that all callbacks have fired
+            // before OnDisconnectedAsync
+
             try
             {
-                // We wait on abort to complete, this is so that we can guarantee that all callbacks have fired
-                // before OnDisconnectedAsync
-
-                try
-                {
-                    // Ensure the connection is aborted before firing disconnect
-                    await connection.AbortAsync();
-                }
-                catch (Exception ex)
-                {
-                    _logger.AbortFailed(ex);
-                }
-
-                using (var scope = _serviceScopeFactory.CreateScope())
-                {
-                    var hubActivator = scope.ServiceProvider.GetRequiredService<IHubActivator<THub>>();
-                    var hub = hubActivator.Create();
-                    try
-                    {
-                        InitializeHub(hub, connection);
-                        await hub.OnDisconnectedAsync(exception);
-                    }
-                    finally
-                    {
-                        hubActivator.Release(hub);
-                    }
-                }
+                // Ensure the connection is aborted before firing disconnect
+                await connection.AbortAsync();
             }
             catch (Exception ex)
             {
-                _logger.ErrorInvokingHubMethod("OnDisconnectedAsync", ex);
-                throw;
+                Log.AbortFailed(_logger, ex);
             }
+
+            await _dispatcher.OnDisconnectedAsync(connection, exception);
```

Comment:
Why did you move this out of the try catch block?

Reply:
Because... uh... shut up. That's why.

Reply:
Actually, the try catch for that code is now in the dispatcher. It was always a log-and-rethrow catch, so it's fine there.

Reply:
We should try catch here now that we can't trust what it does